src/Components/Server/src/Circuits/RemoteJSRuntime.cs

```diff
@@ -9,15 +9,23 @@ namespace Microsoft.AspNetCore.Components.Server.Circuits
 {
     internal class RemoteJSRuntime : JSRuntimeBase
     {
-        private readonly IClientProxy _clientProxy;
+        private IClientProxy _clientProxy;
 
-        public RemoteJSRuntime(IClientProxy clientProxy)
+        public RemoteJSRuntime()
+        {
+        }
+
+        internal void Initialize(IClientProxy clientProxy)
         {
             _clientProxy = clientProxy ?? throw new ArgumentNullException(nameof(clientProxy));
         }
 
         protected override void BeginInvokeJS(long asyncHandle, string identifier, string argsJson)
         {
+            if (_clientProxy == null)
+            {
+                throw new InvalidOperationException("The JavaScript runtime has not been initialized at this point.");
+            }
```

Comment:
It looks like this is the error that someone will see if they try to use the JS Runtime without a circuit (say during pre-rendering). As such should probably make this better:

```
The JavaScript runtime is not available during prerendering. 
```

Reply:
I changed it, don't know why it isn't showing up.

Reply:
Or to be super explicit:

```
The JavaScript runtime is not available during prerendering, because the page has not yet loaded in the browser. Prerendered components must wrap any JavaScript interop calls in conditional logic to ensure those interop calls are not attempted during prerendering.
```